docs/markdown/Pkgconfig-module.md

```diff
@@ -59,3 +61,38 @@ provided for all required fields of the pc file:
 - `install_dir` is set to `pkgconfig` folder in the same location than the provided library.
 - `description` is set to the project's name followed by the library's name.
 - `name` is set to the library's name.
+
+### Implicit dependencies
+
+The exact rules followed to find dependencies that are implicitly added into the
+pkg-config file have evolved over time. Here are the rules as of Meson *0.49.0*,
+previous versions might have slightly different behaviour.
+
+- Not found libraries or dependencies are ignored.
+- Libraries and dependencies are private by default (i.e. added into
+  `Requires.private:` or `Libs.private:`) unless they are explicitly added in
+  `libraries` or `requires` keyword arguments, or is the main library (first
+  positional argument).
+- Libraries and dependencies will be de-duplicated, if they are added in both
+  public and private (e.g `Requires:` and `Requires.private:`) it will be removed
+  from the private list.
+- Shared libraries (i.e. `shared_library()` and **NOT** `library()`) add only
```

Comment:
I don't remember if this is true for `library()` when building with default shared libraries. We should check this.

Reply:
Thanks for reminding me. 

I *think* the rational was that we want the .pc files to be the same regardless of default library type.

Reply:
Yes that's the rational IIRC. It makes sense, even if it's not obvious at all, so it's good that it finally gets documented at least.